From f5fcd57957ddfc7310c66918c81e16ca7a0c56f2 Mon Sep 17 00:00:00 2001
From: Henning Rogge <hrogge@gmail.com>
Date: Wed, 6 Dec 2017 17:44:17 +0100
Subject: [PATCH] info: do not block when sending data

Even though checks are done that we can actually send, the send
can still block.

Reported-By: Joe Ayers <joe@ayerscasa.com>
Signed-off-by: Henning Rogge <hrogge@gmail.com>
Signed-off-by: Ferry Huberts <ferry.huberts@pelagic.nl>
---
 lib/info/olsrd_info.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/info/olsrd_info.c b/lib/info/olsrd_info.c
index 702bddf63..77100d0e7 100644
--- a/lib/info/olsrd_info.c
+++ b/lib/info/olsrd_info.c
@@ -324,7 +324,7 @@ static void send_status_no_retries(const char * req, bool add_headers, int the_s
     }
   }
 
-  (void) send(the_socket, abuf.buf, abuf.len, 0);
+  (void) send(the_socket, abuf.buf, abuf.len, MSG_DONTWAIT);
   close(the_socket);
   abuf_free(&abuf);
 }
@@ -368,7 +368,7 @@ static void write_data(void *unused __attribute__((unused))) {
       continue;
     }
 
-    result = send(outbuffer.socket[i], outbuffer.buffer[i] + outbuffer.written[i], outbuffer.size[i] - outbuffer.written[i], 0);
+    result = send(outbuffer.socket[i], outbuffer.buffer[i] + outbuffer.written[i], outbuffer.size[i] - outbuffer.written[i], MSG_DONTWAIT);
     if (result > 0) {
       outbuffer.written[i] += result;
     }
